### PR TITLE
[dv] Improve sim.py error reporting

### PR DESCRIPTION
**Pull request by GregAC**
This fixes up a few frustrations I've had with the way the regression reporting works. We could clean things up further and improve the python but we're a bit constrained by the way `compare_trace_csv` works from RISCV DV. It outputs to a named logfile and directly reports [PASSED] or [FAILED] into the log so we can't alter how that works nor easily redirect its output somewhere unless we use a temp file for it to write to and we read back from it. I think we're better off changing `compare_trace_csv` in RISCV DV to be more flexible for reporting or reimplementing it on our side.

* Handle missing log files with error messages rather than terminating
  on unhandled exceptions
* Output potential failure causes from sim log file into regression log
* Alter per test output to make it clearer what line corresponds to what
  test
* Only output [PASSED] or [FAILED] a single time per test
* Don't output [PASSED] where sim log is good but ISS comparison is not

### CODE REVIEW

File: dv/uvm/core_ibex/sim.py [hunk 1]
```diff
@@ -380,33 +383,48 @@ def compare_test_run(test, idx, iss, output_dir, report):
...
+                report_fd.write(line)
+            report_fd.write('[FAILED]: sim error seen\n')
+
+            return False
 
-    if no_post_compare:
-        return True
+        if no_post_compare:
+            report_fd.write('[PASSED]\n')
+            return True
```
Comment (rswarbrick) [inline dv/uvm/core_ibex/sim.py:409]: Nit: I don't think this needs to be in the `with` block.

**Review by rswarbrick — APPROVED** [review]
Much better, thanks!

File: dv/uvm/core_ibex/sim.py (hunk 1 above)
Comment (GregAC) [reply dv/uvm/core_ibex/sim.py:409]: It needs access to `report_fd` that the `with` opens. Unless you very quick off the mark and saw my initial version that needlessly nested a second `with` here?

Comment (rswarbrick) [reply dv/uvm/core_ibex/sim.py:409]: Ah, no. I'm just being stupid. Looks good to me!

**Review by udinator — APPROVED** [review]
lgtm!

**Pull request merged**